src/main/java/eu/yals/ui/err/NotFoundView.java

```diff
@@ -0,0 +1,113 @@
+    private final Span subTitle = new Span();
+    private final Image image = new Image();
+
+    public NotFoundView() {
+        init();
+        add(title, subTitle, image);
+        this.setAlignItems(Alignment.CENTER);
+    }
+
+    private void init() {
+        this.setId(IDs.VIEW_ID);
+
+        title.setText("404 - No Such Page Exception");
+        subTitle.setText("We don't have such page at our site. Really.");
+
+        image.setSrc("images/404.jpg");
+        image.setAlt("Error 404 Image");
+    }
+
+    @Override
+    public int setErrorParameter(BeforeEnterEvent event, ErrorParameter<NotFoundException> parameter) {
+        String path = event.getLocation().getPath();
+        boolean identNotFound = path.equals(Endpoint.TNT.SLASH_IDENT);
+        if (identNotFound) {
+            if (AppUtils.clientWantsJson(VaadinRequest.getCurrent())) {
+                VaadinResponse.getCurrent().setHeader(Header.LOCATION, api404Endpoint(event));
+                return 302;
+            } else {
+                subTitle.setText("We don't have long link that match your short link. " +
+                        "Make sure you copypasted it fully and without extra characters");
+                return 404;
+            }
+        }
+        if (StringUtils.isBlank(path)) {
+            event.rerouteTo("/");
+            return 302;
+        } else {
+            boolean isIdentValid = path.matches(IdentGenerator.VALID_IDENT_PATTERN);
+            if (isIdentValid) {
+                List<String> param = new ArrayList<>();
+                param.add(path);
+                event.rerouteTo(Endpoint.TNT.SLASH_IDENT, param);
+                return 302;
+            } else {
+                if (AppUtils.clientWantsJson(VaadinRequest.getCurrent())) {
+                    VaadinResponse.getCurrent().setHeader(Header.LOCATION, Endpoint.Api.PAGE_404);
+                    return 302;
+                }
+                //not a ident
+                if (isApiRequest(path)) {
+                    //api call
+                    VaadinResponse.getCurrent().setHeader(Header.LOCATION, api404Endpoint(event));
+                    return 302;
+                } else {
+                    VaadinResponse.getCurrent().setHeader(Header.LOCATION, Endpoint.UI.PAGE_404);
+                    return 404;
+                }
+            }
+        }
+    }
+
+    private boolean isApiRequest(String path) {
+        return path.startsWith("api");
+    }
+
+    private String api404Endpoint(BeforeEnterEvent event) {
+        String method = VaadinRequest.getCurrent().getMethod();
+        String path = event.getLocation().getPath();
+
+        try {
+            return String.format("%s?method=%s&path=%s", Endpoint.Api.PAGE_404, method,
+                    URLEncoder.encode(path, "UTF-8"));
+        } catch (UnsupportedEncodingException e) {
+            return Endpoint.Api.PAGE_404;
+        }
+    }
+
+    public static class IDs {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The utility class name 'IDs' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/gh/yadevee/yals/pullRequest?prid=5392475)